cli/internal/cmd/spinner.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	spinnerStates = []string{"⣾", "⣽", "⣻", "⢿", "⡿", "⣟", "⣯", "⣷"}
+	dotsStates    = []string{".", "..", "..."}
+)
+
+type Spinner struct {
+	out      *cobra.Command
+	text     string
+	showDots bool
+	delay    time.Duration
+	wg       *sync.WaitGroup
+	stop     int32
+}
+
+func NewSpinner(c *cobra.Command, text string, showDots bool) *Spinner {
+	return &Spinner{
+		out:      c,
+		text:     text,
+		showDots: showDots,
+		wg:       &sync.WaitGroup{},
+		delay:    100 * time.Millisecond,
+		stop:     0,
+	}
+}
+
+func (s *Spinner) Start() {
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+	out:
+		for {
+			for i := 0; i < len(spinnerStates); i++ {
+				if atomic.LoadInt32(&s.stop) == 0 {
+					dotsState := ""
+					if s.showDots {
+						dotsState = dotsStates[i%len(dotsStates)]
+					}
+					state := fmt.Sprintf("\r%s %s%s", spinnerStates[i], s.text, dotsState)
+					s.out.Print(state)
+					time.Sleep(s.delay)
+				} else {
+					break out
+				}
```

Comment:
Go prefers early returns instead of using else blocks. You can add the break condition to the beginning of your loop like this:
```go
			if atomic.LoadInt32(&s.stop) != 0 {
				break
			}
```